kafka-connect-cassandra/src/main/scala/com/datamountaineer/streamreactor/connect/cassandra/utils/KeyUtils.scala

```diff
@@ -28,6 +27,14 @@ object KeyUtils {
     * @return
     */
   def keysFromStruct(struct: Struct, schema: Schema, fieldNames: Seq[String]): Seq[Object] =
-    keysFromJson(new SimpleJsonConverter().fromConnectData(schema, struct).toString, fieldNames)
+    fieldNames.map(getKeyFromStruct(struct, _))
```

Comment:
we have 
`com.datamountaineer.streamreactor.connect.converters.source.KeyExtractor::extract(Struct, Vector[String])`

if you split by `.` you have the vector[string] and the struct is there already.